### PR TITLE
Added end-to-end test for states and transitions

### PR DESCRIPTION
**Pull request by OleksandrChaika**

### CODE REVIEW

File: src/FSM/States/Downloading.cpp [hunk 1]
```diff
@@ -58,8 +58,7 @@ namespace sua {
             Logger::info("Bundle version (spec): '{}'", ctx.desiredState.bundleVersion);
             Logger::info("Bundle version (file): '{}'", bundleVersion);
 
-            if(ctx.bundleChecker->isBundleVersionConsistent(
```
Comment (magdalenakusz) [inline src/FSM/States/Downloading.cpp:61]: just a question: is this method not used any more? can it be deleted if the check is performed directly?

**Review by magdalenakusz — COMMENTED** [review]
Looks good for me (code level) just had one question, if the method could be now removed - not sure myself.

**Review by magdalenakusz — APPROVED** [review]
LGTM

File: src/FSM/States/Downloading.cpp (hunk 1 above)
Comment (OleksandrChaika) [reply src/FSM/States/Downloading.cpp:61]: Removed

**Pull request merged**